lib/Sema/CSDiag.cpp

```diff
@@ -3443,15 +3443,37 @@ Expr *FailureDiagnosis::typeCheckChildIndependently(
       if (FT->isAutoClosure())
         convertType = FT->getResult();
 
-    // Replace archetypes and type parameters with UnresolvedType.
-    convertType = replaceTypeParametersWithUnresolved(convertType);
-    convertType = replaceTypeVariablesWithUnresolved(convertType);
-    
+    bool shouldNullify = false;
```

Comment:
I moved both calls down after decision to nullify because we need original types to figure what was what to decide nullification.